SLHCUpgradeSimulations/Phase2TrackerDTCCablingMap/test/DTCCablingMapProducer_write.py

```diff
@@ -0,0 +1,54 @@
+import FWCore.ParameterSet.Config as cms
+
+process = cms.Process("DTCCablingMapTestProducerWriteDEMO")
+
+process.load("FWCore.MessageLogger.MessageLogger_cfi")
+
+
+process.load('Configuration.Geometry.GeometryExtended2023D17Reco_cff')
```

Comment:
Hello Gabrielle :)

There is no CSV in the cabling map package anymore, I removed them following the discussion above in this thread. Eventually some of these reference CSVs could will be put into the cms-data repository, but it may be useless to duplicate in cms-data a file which can be retrieved in tkLayout. I cannot judge the longevity of layouts in tkLayout: if they may be erased in the future maybe cms-data is a good place to store the important CSVs.

The cabling map is agnostic to the version of the geometry the cabling corresponds to and it sould remain that way, because the cabling map does not hard-code the cabling: all the cabling is in the data it contains. Adding checks of any sort would make it fragile and complicated, since the input CSV file does not bear any information on which geometry it corresponds to. You would need either a filename-based heuristic (horrible) or you would need to become dependent from the geometry packages (the least dependence you have, the better imo).

Ultimately, the correct choice of which cabling to load will be up to the person responsible for uploading the payload into the database (for central and default workflow jobs) or the user of CMSSW when a different cabling map is used (for private jobs).

Nevertheless, since we are temporarily circumventing the limitations implied by the use of a single global tag for all Phase-2 upgrade modelizations (we will only include the OT map to start with, because that map is "theoretically frozen"), I think @ghugo83 comment underlines that there is already a tension between what people need and what the current organization of tags in the database can provide. @mmusich, do you think there is any hope the single-global-tag policy will change in the future? After all the global tag is about a known detector configuration, while upgrade activities are dealing with literally different CMS detectors.

Reply:
> Nevertheless, since we are temporarily circumventing the limitations implied by the use of a single global tag for all Phase-2 upgrade modelizations (we will only include the OT map to start with, because that map is "theoretically frozen"), I think @ghugo83 comment underlines that there is already a tension between what people need and what the current organization of tags in the database can provide. @mmusich, do you think there is any hope the single-global-tag policy will change in the future? After all the global tag is about a known detector configuration, while upgrade activities are dealing with literally different CMS detectors.

I think the decision is not up to me, but rather up to CMS in general and this is the ideal topic to be discussed at an AlCa/DB meeting @tocheng @ggovi .
By the way nothing prevents you to have different cabling maps in GT under different labels, you will just need to be sure when you produce events, you pick the desired one.

Reply:
Hi Luigi :)

- On the csv: yes it is good to have them in cms-data. The cmssw-models tab on tkLayout website is frozen, but the other links are not, so is good to keep a copy of what has been used.
You do not need all these csv though, only the latest OT cabling map is fine, no? In theory you should have only 1 csv, no? What do you think?

- On the geo / cabling map match: Yes it is bound to the user to know what they are doing. But if the naming and comments and readme are explicit, it avoids issues (like the one you did yourself, imagine people who did not work specifically on this topic) :)
I was not thinking of some core tool, more just:
- renaming the csv which you want to store, to contain the name of the tracker it was designed for (T6, T14 ...)
- add a comment in SLHCUpgradeSimulations/Phase2TrackerDTCCablingMap/test/DTCCablingMapProducer_write.py about that point to warn people.
- add a readme in SLHCUpgradeSimulations/Phase2TrackerDTCCablingMap/test/ on the cabling maps.

What do you think? :)

Reply:
Regarding global tags: the other side of the argument is that we clearly have no personpower to maintain O(20) global tag queues for Phase 2, if you want to have one for each detector version currently supported.

Reply:
Hi @ghugo83, I actually think it's not needed anymore to import the geometry pset (I needed it when I was adding the sensor->DTC mapping, but that was removed). I think that the line can be removed from DTCCablingMapProducer_write.py .

I will add a brief README recommending to take care of the tracker model version.

@kpedro88, I see. Do you think the workload requirements for maintaining the GT could be eased by increased automation (I'm asking out of ignorance)?

Reply:
That's really a question for AlCa.